modules/network/src/main/java/org/apache/ignite/internal/network/direct/stream/DirectByteBufferStreamImplV1.java

```diff
@@ -1049,6 +1166,80 @@ public BitSet readBitSet() {
         return arr != null ? BitSet.valueOf(arr) : null;
     }
 
+    @Override
+    public ByteBuffer readByteBuffer() {
+        byte[] bytes;
+
+        switch (byteBufferState) {
+            case 0:
+                byteBufferFlag = readByte();
+
+                boolean isNull = byteBufferFlag == 0;
+
+                if (!lastFinished || isNull) {
+                    return null;
+                }
+
+                byteBufferState++;
+
+                //noinspection fallthrough
+            case 1:
+                if ((byteBufferFlag & 4) != 0) {
+                    byteBufferPosition = 0;
+                } else {
+                    byteBufferPosition = readInt();
+                }
+
+                if (!lastFinished) {
+                    return null;
+                }
+
+                byteBufferState++;
+
+                //noinspection fallthrough
+            case 2:
+                if ((byteBufferFlag & 8) != 0) {
+                    byteBufferLimit = -1;
```

Comment:
This seems to be correct, but in `write` method we write `limit` in case if flag is set